### PR TITLE
[NPM] Optimize Failed Connection kprobes

### PR DESCRIPTION
**Pull request by akarpz**
### What does this PR do?

An optimization to https://github.com/DataDog/datadog-agent/pull/25517, this PR better handles TCP failures that trigger the `tcp_done` and/or `tcp_close` syscalls. It improves the synchronization of closed connection flushes between the two and increases the size of the map used to track that state. 

### Motivation

https://datadoghq.atlassian.net/browse/NPM-3297

### CODE REVIEW

File: pkg/network/tracer/connection/tracer.go [hunk 1]
```diff
@@ -534,13 +524,14 @@ func (t *tracer) Remove(conn *network.ConnectionStats) error {
...
-	removeConnection(conn)
+	removeConnectionFromTelemetry(conn)
 
-	// We have to remove the PID to remove the element from the TCP Map since we don't use the pid there
-	t.removeTuple.Pid = 0
 	if conn.Type == network.TCP {
 		// We can ignore the error for this map since it will not always contain the entry
 		_ = t.tcpStats.Delete(t.removeTuple)
+		// We remove the PID from the tuple as it is not used in the retransmits map
+		conn.Pid = 0
```
Comment (hmahmood) [inline pkg/network/tracer/connection/tracer.go:533]: `t.removeTuple.Pid = 0`

Comment (hmahmood) [reply pkg/network/tracer/connection/tracer.go:533]: Should also restore it afterwards.

File: pkg/network/tracer/connection/failure/matching.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ var (
 	}
 
 	telemetryModuleName = "network_tracer__tcp_failure"
-	mapTTL              = 60 * time.Second.Nanoseconds()
+	mapTTL              = 10 * time.Millisecond.Nanoseconds()
```
Comment (hmahmood) [inline pkg/network/tracer/connection/failure/matching.go:35]: This is drastic increase from before. Is such a big change justified?

File: pkg/network/tracer/connection/tracer.go (hunk 1 above)
Comment (akarpz) [reply pkg/network/tracer/connection/tracer.go]: added

**Review by hmahmood — APPROVED** [review]

File: pkg/network/tracer/connection/failure/matching.go (hunk 2 above)
Comment (akarpz) [reply pkg/network/tracer/connection/failure/matching.go:35]: discussed offline and we are going to leave this and set the frequency of map cleaner runs to 1s

**Pull request merged**